jmx/jmx-mbeanserver-platform/src/main/java/org/apache/aries/jmx/mbean_server/platform/impl/Activator.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.aries.jmx.mbean_server.platform.impl;
+
+import java.lang.management.ManagementFactory;
+import java.util.Hashtable;
+
+import javax.management.Attribute;
+import javax.management.AttributeList;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+
+public class Activator implements BundleActivator {
+
+    private final Object registrationLock = new Object();
+    private ServiceRegistration<MBeanServer> registration;
+
+    @Override
+    public void start(BundleContext context) throws Exception {
```

Comment:
Yes, that makes sense. I tried to bet the reactor build working. It requires Java 8 and then the issue with with the pax-exam tests that fail because of https://ops4j1.jira.com/browse/PAXURL-352 . I patched in the latest pax-url version and added a missing dependency.

```diff
diff --git a/jmx/jmx-itests/pom.xml b/jmx/jmx-itests/pom.xml
index 10f920f17..50f8873e0 100644
--- a/jmx/jmx-itests/pom.xml
+++ b/jmx/jmx-itests/pom.xml
@@ -38,7 +38,7 @@
 
      <properties>
         <exam.version>3.4.0</exam.version>
-        <url.version>1.6.0</url.version>
+        <url.version>2.5.3</url.version>
     </properties>
 
     <dependencies>
@@ -181,6 +181,12 @@
             <version>${url.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.geronimo.specs</groupId>
+            <artifactId>geronimo-atinject_1.0_spec</artifactId>
+            <version>1.0</version>
+            <scope>test</scope>
+        </dependency>
         <dependency>
             <groupId>ch.qos.logback</groupId>
             <artifactId>logback-core</artifactId>
diff --git a/jmx/jmx-whiteboard/pom.xml b/jmx/jmx-whiteboard/pom.xml
index 1103faacc..dd708ca9a 100644
--- a/jmx/jmx-whiteboard/pom.xml
+++ b/jmx/jmx-whiteboard/pom.xml
@@ -81,7 +81,7 @@
         </aries.osgi.import.service>
 
         <exam.version>3.4.0</exam.version>
-        <url.version>1.6.0</url.version>
+        <url.version>2.5.3</url.version>
     </properties>
 
     <build>
@@ -192,6 +192,12 @@
             <version>${url.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.geronimo.specs</groupId>
+            <artifactId>geronimo-atinject_1.0_spec</artifactId>
+            <version>1.0</version>
+            <scope>test</scope>
+        </dependency>
         <dependency>
             <groupId>org.ops4j.pax.tinybundles</groupId>
             <artifactId>tinybundles</artifactId>
```

With this I get the tests running but most of them fail

```
[ERROR] Errors: 
[ERROR]   BundleStateMBeanTest.testAttributeChangeNotifications » ClassNotFound org.osgi.jmx.framework.BundleStateMBean
[ERROR]   BundleStateMBeanTest.testBundleIDsAttribute » ClassNotFound org.osgi.jmx.framework.BundleStateMBean
[ERROR]   BundleStateMBeanTest.testHeaderLocalization » ClassNotFound org.osgi.jmx.framework.BundleStateMBean
[ERROR]   BundleStateMBeanTest.testMBeanInterface » ClassNotFound org.osgi.jmx.framework.BundleStateMBean
[ERROR]   BundleStateMBeanTest.testObjectName » ClassNotFound org.osgi.jmx.framework.BundleStateMBean
[ERROR]   FrameworkMBeanTest.testGetDependencyClosure » ClassNotFound org.osgi.jmx.framework.FrameworkMBean
[ERROR]   FrameworkMBeanTest.testGetProperty » ClassNotFound org.osgi.jmx.framework.FrameworkMBean
[ERROR]   FrameworkMBeanTest.testMBeanInterface » ClassNotFound org.osgi.jmx.framework.FrameworkMBean
[ERROR]   FrameworkMBeanTest.testObjectName » ClassNotFound org.osgi.jmx.framework.FrameworkMBean
[ERROR]   FrameworkMBeanTest.testRefreshBundleAndWait » ClassNotFound org.osgi.jmx.framework.FrameworkMBean
[ERROR]   FrameworkMBeanTest.testRefreshBundlesAndWait » ClassNotFound org.osgi.jmx.framework.FrameworkMBean
[ERROR]   FrameworkMBeanTest.testRefreshBundlesAndWait2 » ClassNotFound org.osgi.jmx.framework.FrameworkMBean
[ERROR]   PackageStateMBeanTest.doSetUp:59->AbstractIntegrationTest.waitForMBean:172->AbstractIntegrationTest.waitForMBean:189 » Runtime osgi.core:type=packageState,version=1.5 mbean is not available after waiting 20 seconds
[ERROR]   PackageStateMBeanTest.doSetUp:59->AbstractIntegrationTest.waitForMBean:172->AbstractIntegrationTest.waitForMBean:189 » Runtime osgi.core:type=packageState,version=1.5 mbean is not available after waiting 20 seconds
[ERROR]   ServiceStateMBeanTest.testAttributeChangeNotifications » ClassNotFound org.osgi.jmx.framework.ServiceStateMBean
[ERROR]   ServiceStateMBeanTest.testGetServiceAndGetProperty » ClassNotFound org.osgi.jmx.framework.ServiceStateMBean
[ERROR]   ServiceStateMBeanTest.testGetServiceIds » ClassNotFound org.osgi.jmx.framework.ServiceStateMBean
[ERROR]   ServiceStateMBeanTest.testListServices » ClassNotFound org.osgi.jmx.framework.ServiceStateMBean
[ERROR]   ServiceStateMBeanTest.testListServicesSelectiveItems » ClassNotFound org.osgi.jmx.framework.ServiceStateMBean
[ERROR]   ServiceStateMBeanTest.testMBeanInterface » ClassNotFound org.osgi.jmx.framework.ServiceStateMBean
[ERROR]   ServiceStateMBeanTest.testObjectName » ClassNotFound org.osgi.jmx.framework.ServiceStateMBean
[ERROR]   ServiceStateMBeanTest.testServicePropertiesInListServices » ClassNotFound org.osgi.jmx.framework.ServiceStateMBean
[ERROR]   BundleWiringStateMBeanTest.testCurrentWiringClosure » ClassNotFound org.osgi.jmx.framework.wiring.BundleWiringStateMBean
[ERROR]   BundleWiringStateMBeanTest.testGetCurrentRevisionDeclaredCapabilities » ClassNotFound org.osgi.jmx.framework.wiring.BundleWiringStateMBean
[ERROR]   BundleWiringStateMBeanTest.testGetCurrentRevisionDeclaredRequirements » ClassNotFound org.osgi.jmx.framework.wiring.BundleWiringStateMBean
[ERROR]   BundleWiringStateMBeanTest.testGetCurrentWiring » ClassNotFound org.osgi.jmx.framework.wiring.BundleWiringStateMBean
[ERROR]   BundleWiringStateMBeanTest.testGetRevisionsDeclaredCapabilities » ClassNotFound org.osgi.jmx.framework.wiring.BundleWiringStateMBean
[ERROR]   BundleWiringStateMBeanTest.testGetRevisionsDeclaredRequirements » ClassNotFound org.osgi.jmx.framework.wiring.BundleWiringStateMBean
[ERROR]   BundleWiringStateMBeanTest.testObjectName » ClassNotFound org.osgi.jmx.framework.wiring.BundleWiringStateMBean
[ERROR]   BundleWiringStateMBeanTest.testRevisionsWiring » ClassNotFound org.osgi.jmx.framework.wiring.BundleWiringStateMBean
[ERROR]   BundleWiringStateMBeanTest.testRevisionsWiringClosure » ClassNotFound org.osgi.jmx.framework.wiring.BundleWiringStateMBean
[ERROR]   PermissionAdminMBeanTest.testMBeanInterface:54 » ClassNotFound org.osgi.jmx.service.permissionadmin.PermissionAdminMBean
[INFO] 
[ERROR] Tests run: 36, Failures: 0, Errors: 32, Skipped: 2
```

I looked into the Pax-Exam test output but I don't understand what is going wrong. Some snippets seem to suggest that not all the Aries JMX bundles resolve but I'm not sure how to figure out the root cause. Would you have an idea?

```
10:38:17.713 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.exam.link'}'}
10:38:17.713 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.exam.inject.link'}'}
10:38:17.714 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.extender.service.link'}'}
10:38:17.714 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.osgi.compendium.link'}'}
10:38:17.714 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.logging.api.link'}'}
10:38:17.714 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.base.link'}'}
10:38:17.715 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.swissbox.core.link'}'}
10:38:17.715 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.swissbox.extender.link'}'}
10:38:17.716 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.swissbox.framework.link'}'}
10:38:17.716 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.swissbox.lifecycle.link'}'}
10:38:17.716 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.swissbox.tracker.link'}'}
10:38:17.717 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.apache.geronimo.specs.atinject.link'}'}
10:38:17.717 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@2) JUnitBundlesOption{url=link:classpath:META-INF/links/org.ops4j.pax.tipi.junit.link}
10:38:17.717 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.tipi.hamcrest.core.link'}'}
10:38:17.717 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) UrlProvisionOption{url='RawUrlReference{url='link:classpath:META-INF/links/org.ops4j.pax.exam.invoker.junit.link'}'}
10:38:17.718 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) MavenArtifactUrlReference{groupId='org.osgi', artifactId='org.osgi.compendium', version='4.3.1', type='null'}
10:38:17.718 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) MavenArtifactUrlReference{groupId='org.apache.aries', artifactId='org.apache.aries.util', version='1.0.0', type='null'}
10:38:17.719 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) MavenArtifactUrlReference{groupId='org.apache.felix', artifactId='org.apache.felix.configadmin', version='1.8.0', type='null'}
10:38:17.719 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) MavenArtifactUrlReference{groupId='org.apache.aries.jmx', artifactId='org.apache.aries.jmx', version='1.1.6-SNAPSHOT', type='null'}
10:38:17.719 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) MavenArtifactUrlReference{groupId='org.apache.aries.jmx', artifactId='org.apache.aries.jmx.core.whiteboard', version='1.1.7-SNAPSHOT', type='null'}
10:38:17.719 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) MavenArtifactUrlReference{groupId='org.apache.aries.jmx', artifactId='org.apache.aries.jmx.api', version='1.1.6-SNAPSHOT', type='null'}
10:38:17.720 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) MavenArtifactUrlReference{groupId='org.apache.aries.jmx', artifactId='org.apache.aries.jmx.whiteboard', version='1.1.6-SNAPSHOT', type='null'}
10:38:17.720 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) MavenArtifactUrlReference{groupId='org.apache.aries.testsupport', artifactId='org.apache.aries.testsupport.unit', version='2.0.0-SNAPSHOT', type='null'}
10:38:17.720 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) UrlProvisionOption{url='RawUrlReference{url='file:/tmp/ops4j-store-anonymous-2226271183744275890/tinybundles_9469795b111d7975b864956cc60e1f9e3531f082.bin'}'}
10:38:17.721 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - + Install (start@3) UrlProvisionOption{url='RawUrlReference{url='file:/tmp/ops4j-store-anonymous-5461496698858606604/tinybundles_2be0ea4f48bd9f66c6ee00f112829b0522c7884f.bin'}'}
10:38:17.721 [main] DEBUG o.o.p.e.forked.ForkedTestContainer - Jump to startlevel [5]
org.ops4j.pax.logging.pax-logging-api[org.ops4j.pax.logging.internal.Activator] : Enabling SLF4J API support.
org.ops4j.pax.logging.pax-logging-api[org.ops4j.pax.logging.internal.Activator] : Enabling Jakarta Commons Logging API support.
org.ops4j.pax.logging.pax-logging-api[org.ops4j.pax.logging.internal.Activator] : Enabling Log4J API support.
org.ops4j.pax.logging.pax-logging-api[org.ops4j.pax.logging.internal.Activator] : Enabling Avalon Logger API support.
org.ops4j.pax.logging.pax-logging-api[org.ops4j.pax.logging.internal.Activator] : Enabling JULI Logger API support.
10:38:17.774 [main] ERROR o.o.p.e.forked.ForkedTestContainer - Bundle [19] is not resolved
10:38:17.774 [main] ERROR o.o.p.e.forked.ForkedTestContainer - Bundle [20] is not resolved
10:38:17.774 [main] ERROR o.o.p.e.forked.ForkedTestContainer - Bundle [21] is not resolved
10:38:17.774 [main] ERROR o.o.p.e.forked.ForkedTestContainer - Bundle [22] is not resolved
10:38:17.774 [main] ERROR o.o.p.e.forked.ForkedTestContainer - Bundle [25] is not resolved
```

Reply:
Hi, I am not sure if it is so bad :) 
On your branch using java 8 without changes from your comment above I run:
```
mvn -f jmx/pom.xml -am -pl jmx-itests clean package
```
and the result is:

```
[INFO] --- surefire:3.1.2:test (default-test) @ org.apache.aries.jmx.itests ---
[INFO] Using auto detected provider org.apache.maven.surefire.junit4.JUnit4Provider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.aries.jmx.test.blueprint.BlueprintMBeanTest
^[c^[c[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.130 s -- in org.apache.aries.jmx.test.blueprint.BlueprintMBeanTest
[INFO] Running org.apache.aries.jmx.cm.ConfigurationAdminMBeanTest
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 1.938 s -- in org.apache.aries.jmx.cm.ConfigurationAdminMBeanTest
[INFO] Running org.apache.aries.jmx.provisioning.ProvisioningServiceMBeanTest
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 1.457 s -- in org.apache.aries.jmx.provisioning.ProvisioningServiceMBeanTest
[INFO] Running org.apache.aries.jmx.permissionadmin.PermissionAdminMBeanTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.442 s -- in org.apache.aries.jmx.permissionadmin.PermissionAdminMBeanTest
[INFO] Running org.apache.aries.jmx.framework.FrameworkMBeanTest
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 10.81 s -- in org.apache.aries.jmx.framework.FrameworkMBeanTest
[INFO] Running org.apache.aries.jmx.framework.wiring.BundleWiringStateMBeanTest
[INFO] Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.500 s -- in org.apache.aries.jmx.framework.wiring.BundleWiringStateMBeanTest
[INFO] Running org.apache.aries.jmx.framework.BundleStateMBeanTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 8.550 s -- in org.apache.aries.jmx.framework.BundleStateMBeanTest
[INFO] Running org.apache.aries.jmx.framework.ServiceStateMBeanTest
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 10.86 s -- in org.apache.aries.jmx.framework.ServiceStateMBeanTest
[INFO] Running org.apache.aries.jmx.framework.PackageStateMBeanTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.389 s -- in org.apache.aries.jmx.framework.PackageStateMBeanTest
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 36, Failures: 0, Errors: 0, Skipped: 2
...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Apache Aries JMX parent 1.1.6-SNAPSHOT ............. SUCCESS [  0.778 s]
[INFO] Apache Aries JMX API 1.1.6-SNAPSHOT ................ SUCCESS [  0.790 s]
[INFO] Apache Aries JMX Core 1.1.9-SNAPSHOT ............... SUCCESS [  1.568 s]
[INFO] Apache Aries JMX Core via Whiteboards 1.1.7-SNAPSHOT SUCCESS [  0.193 s]
[INFO] Apache Aries JMX Bundle 1.1.6-SNAPSHOT ............. SUCCESS [  0.261 s]
[INFO] Apache Aries JMX Blueprint Bundle 1.1.6-SNAPSHOT ... SUCCESS [  0.167 s]
[INFO] Apache Aries JMX integration tests 1.1.6-SNAPSHOT .. SUCCESS [ 42.556 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  46.792 s
[INFO] Finished at: 2024-11-28T20:47:00+01:00
```

With command:
```
mvn -f jmx/pom.xml clean package
```
the run is also green:
```
[INFO] Reactor Summary:
[INFO] 
[INFO] Apache Aries JMX parent 1.1.6-SNAPSHOT ............. SUCCESS [  0.842 s]
[INFO] Apache Aries JMX API 1.1.6-SNAPSHOT ................ SUCCESS [  0.733 s]
[INFO] Apache Aries JMX Blueprint API 1.2.1-SNAPSHOT ...... SUCCESS [  0.403 s]
[INFO] Apache Aries JMX Core 1.1.9-SNAPSHOT ............... SUCCESS [  1.157 s]
[INFO] Apache Aries JMX Core via Whiteboards 1.1.7-SNAPSHOT SUCCESS [  0.176 s]
[INFO] Apache Aries Whiteboard support for JMX DynamicMBean services 1.2.1-SNAPSHOT SUCCESS [  0.736 s]
[INFO] Apache Aries JMX Blueprint Core 1.2.1-SNAPSHOT ..... SUCCESS [  0.408 s]
[INFO] Apache Aries JMX Bundle 1.1.6-SNAPSHOT ............. SUCCESS [  0.232 s]
[INFO] Apache Aries JMX Blueprint Bundle 1.1.6-SNAPSHOT ... SUCCESS [  0.149 s]
[INFO] Apache Aries JMX integration tests 1.1.6-SNAPSHOT .. SUCCESS [ 36.940 s]
[INFO] Apache Aries JMX MBeanServer (Platform) 0.1.0-SNAPSHOT SUCCESS [  0.099 s]
[INFO] Apache Aries JMX 1.1.6-SNAPSHOT .................... SUCCESS [  0.051 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  42.394 s
[INFO] Finished at: 2024-11-28T20:50:09+01:00
```

Reply:
Hm, I am not sure what is going on ... . I think there are two things at play here:

1. You already have all the needed jar downloaded locally. A second test run works for me as well without my pax-url version upgrade. Can you try building with an empty local Maven repository?
2. The pax-exam ITs definitely fail with the same problems for me. I am not sure what to do, except trying to upgrade other bits of pax-exam, I recall that newer versions have better error output.

Reply:
I added a simple GH actions workflow to my fork and created a PR with it - https://github.com/apache/aries/pull/298. 

Without the fix this fails with the error I initially noticed - https://github.com/rombert/aries/actions/runs/12086375048/job/33705541718 .

After I pushed the fix the Pax-Exam tests still fail - https://github.com/rombert/aries/actions/runs/12086416989/job/33705665437.

Reply:
I see it also on my branch and PR https://github.com/apache/aries/pull/297
After cleaning maven repo I have the same problems

Reply:
I was able to have green build locally with some hacky way...
```
$ rm -r ~/.m2/repository/
$ git llog | head -5
662919507 2024-11-26 16:45:57 +0100 Christian Schneider (HEAD -> trunk, origin/trunk, origin/HEAD) Merge pull request #295 from rombert/fix/jmx-parent-update
7ad81cb47 2024-11-26 15:41:01 +0100 Robert Munteanu jmx: use Aries parent 2.1.1
e93c8014b 2023-11-14 22:16:29 +0100 Guillaume Nodet Fix javadoc
aea7d9056 2023-11-07 20:56:47 +0100 JB Onofré [maven-release-plugin] prepare for next development iteration
e9086cbe9 2023-11-07 20:56:32 +0100 JB Onofré (tag: spifly-1.3.7) [maven-release-plugin] prepare release spifly-1.3.7
$ mvn -pl jmx/jmx-itests clean install
...
$ git checkout e93c8014b
...
$ mvn -am -pl jmx/jmx-bundle clean install
...
$ git checkout trunk
...
# update pax url to 2.5.3 in itest pom.xml
...
$ mvn -pl jmx/jmx-itests clean install # note: no -am
```
It looks like jmx-bundle built with newer parent makes the tests fail.
I see the only one significant difference in jmx-bundle jars -
```
Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.6))"
```
is changed to 
```
Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.8))"
```

Reply:
I prepared the workflow for JMX with fixes for itest https://github.com/apache/aries/pull/297
I will be happy to merge it before your changes - please let me know if it unblocks your tests in this PR

Reply:
Thanks for this! I added a small question to that PR, you seem to have downgraded a bundle.

Reply:
Thanks, I will wait for Aries team review in my PR - if there is no review I will merge my PR. Then you should be able to add the tests to this PR.

Reply:
Sounds good, thanks!

Reply:
PR #297 is merged

Reply:
Thanks! I rebased, updated the new bundle version as suggested and will look into adding tests this week.